dali/kernels/signal/moving_mean_square_gpu.cu

```diff
@@ -184,8 +184,10 @@ __global__ void SlidingWindowSum(const SampleDesc<Out, In> *samples, int64_t log
     // Step 3: Compute the output, the sum in window, by subtracting two values of the prefix sum
     // and adding the input value at the current position.
     for (int pos = threadIdx.x; pos < logical_block_sz; pos += blockDim.x) {
-      acc_t<In> out_val = pre(logical_block_in_ptr[pos]) + temp[shm_pos(window + pos)] -
-                          temp[shm_pos(pos)];
+      acc_t<In> x = logical_block_in_ptr[pos];
+      acc_t<In> out_val = pre(x)                          // current element
+                          + temp[shm_pos(window + pos)]   // prefix sum @ pos
+                          - temp[shm_pos(pos + 1)];       // prefix sum @ pos - (window-1)
```

Comment:
Two things worth noting here:
1. `pre` should happen on `acc_t<In>` type, not `In`
2. `pos + 1` for the tail, otherwise we are summing `window+1` samples effectively